app/src/main/kotlin/com/intive/picover/images/repository/ImagesRepository.kt

```diff
@@ -0,0 +1,32 @@
+package com.intive.picover.images.repository
+
+import android.net.Uri
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.StorageReference
+import javax.inject.Inject
+import kotlin.coroutines.resumeWithException
+import kotlinx.coroutines.suspendCancellableCoroutine
+
+class ImagesRepository @Inject constructor() {
+	private val storageRef = FirebaseStorage.getInstance().reference.child("image")
+
+	internal suspend fun fetchItems(): List<StorageReference> {
+		return suspendCancellableCoroutine {
+			storageRef.listAll().addOnSuccessListener { listResult ->
+				it.resumeWith(Result.success(listResult.items))
+			}.addOnFailureListener { exception ->
+				it.resumeWithException(exception)
+			}
+		}
+	}
+
+	internal suspend fun downloadUri(storageReference: StorageReference): Uri {
+		return suspendCancellableCoroutine {
+			storageReference.downloadUrl.addOnSuccessListener { uri ->
+				it.resumeWith(Result.success(uri))
```

Comment:
minor: You could replace `it` with something meaningful. You did it two lines above with `uri`.

Reply:
Good point. Thank you.